### PR TITLE
Update to Mapbox Directions `v2.8.0-beta.1`, Mapbox Maps `v10.9.0-rc.1`, Navigation Native `v118.0.0` and Common `v23.1.0-rc.2`.

### PR DESCRIPTION
**Pull request by MaximAlien**
### Description

Update dependencies to Mapbox Directions `v2.8.0-beta.1`, Mapbox Maps `v10.9.0-rc.1`, Navigation Native `v118.0.0` and Common `v23.1.0-rc.2`.

### CODE REVIEW

File: Cartfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-binary "https://api.mapbox.com/downloads/v2/carthage/mapbox-common/MapboxCommon.json" == 23.1.0-beta.2
-binary "https://api.mapbox.com/downloads/v2/carthage/mobile-navigation-native/MapboxNavigationNative.xcframework.json" ~> 116.0
-github "mapbox/mapbox-directions-swift" == 2.8.0-alpha.1
+binary "https://api.mapbox.com/downloads/v2/carthage/mapbox-common/MapboxCommon.json" == 23.1.0-rc.1
+binary "https://api.mapbox.com/downloads/v2/carthage/mobile-navigation-native/MapboxNavigationNative.xcframework.json" ~> 117.0
+github "mapbox/mapbox-directions-swift" == 2.8.0-alpha.2
```
Comment (MaximAlien) [inline Cartfile:3]: Seems like Directions dependency wasn't updated for Carthage in previous release.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,9 +6,9 @@
...
-* MapboxCoreNavigation now requires [MapboxNavigationNative v116._x_](https://github.com/mapbox/mapbox-navigation-native-ios/releases/tag/116.0.0). ([#4183](https://github.com/mapbox/mapbox-navigation-ios/pull/4183))
+* MapboxCoreNavigation now requires [MapboxNavigationNative v117._x_](https://github.com/mapbox/mapbox-navigation-native-ios/releases/tag/117.0.0). ([#4191](https://github.com/mapbox/mapbox-navigation-ios/pull/4191))
 * MapboxNavigation now requires [MapboxSpeech v2._x_](https://github.com/mapbox/mapbox-speech-swift/releases/tag/v2.1.0). ([#4142](https://github.com/mapbox/mapbox-navigation-ios/pull/4142))
-* MapboxNavigation now requires [MapboxMaps v10.9.0-beta.2](https://github.com/mapbox/mapbox-maps-ios/releases/tag/v10.9.0-beta.2). ([#4183](https://github.com/mapbox/mapbox-navigation-ios/pull/4183))
+* MapboxNavigation now requires [MapboxMaps v10.9.0-rc.1](https://github.com/mapbox/mapbox-maps-ios/releases/tag/v10.9.0-rc.1). ([#4191](https://github.com/mapbox/mapbox-navigation-ios/pull/4191))
 
 ### Routing
```
Comment (1ec5) [inline CHANGELOG.md:13]: Suggested changelog entries:

### Location tracking

* Fixed an issue where the user’s location history was sometimes matched to unnavigable paths, leading to unreliable location snapping. ([#4191](https://github.com/mapbox/mapbox-navigation-ios/pull/4191))
* Fixed an issue where the `RoadObjectMatcher.matchOpenLR(location:identifier:)` method would incorrectly match some TPEG OpenLR identifiers. ([#4191](https://github.com/mapbox/mapbox-navigation-ios/pull/4191))

### Routing

* Fixed an issue where some alternative routes did not begin at the user’s current location or did not end at the same location as the main route. ([#4191](https://github.com/mapbox/mapbox-navigation-ios/pull/4191))
* Fixed an issue where some continuous alternative routes were unavailable or not consistently available when the user is traveling at low speed. ([#4191](https://github.com/mapbox/mapbox-navigation-ios/pull/4191))
* Fixed an issue where the values of `RouteLeg.segmentDistances` and `RouteLeg.expectedTravelTime` did not add up to `RouteLeg.distance` and `RouteLeg.expectedTravelTime`, respectively. Each value of `RouteLeg.expectedTravelTime` that immediately follows an intersection now includes the time required to traverse the intersection. ([#4191](https://github.com/mapbox/mapbox-navigation-ios/pull/4191))
* Fixed an issue where some alternative routes were identical to the main route due to suppressing a [rat run](https://en.wikipedia.org/wiki/Rat_running). ([#4191](https://github.com/mapbox/mapbox-navigation-ios/pull/4191))
* Fixed an issue where routes avoided roads with the deprecated tags [`hov=lane`](https://taginfo.openstreetmap.org/tags/hov=lane) and `hov:conditional=lane @ …` as restricted roads. ([#4191](https://github.com/mapbox/mapbox-navigation-ios/pull/4191))

### Guidance instructions

* Turn lane indications now appear even if they slightly differ from what the routing engine would have inferred from the road geometry, For example, turn lanes now appear where the user is expected to use a lane marked as as a right turn lane to make a slight right turn. ([#4191](https://github.com/mapbox/mapbox-navigation-ios/pull/4191))
* Route shields now respect language-specific route numbers in Japan. ([#4191](https://github.com/mapbox/mapbox-navigation-ios/pull/4191))
* Improved the timing of some spoken instructions. ([#4191](https://github.com/mapbox/mapbox-navigation-ios/pull/4191))

**Comment by 1ec5** [comment]
Tests still [deadlock](https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-ios/9729/workflows/33abd3f7-276c-4052-8177-92cb0ba7219c/jobs/89206) even with MapboxCommon v23.1.0-rc.2:

```
2022-10-07 18:08:50.800639+0000 xctest[4776:20756] [Mapbox] [Warning, config_service]: No configuration options found stored on the disk.
2022-10-07 18:08:50.809792+0000 xctest[4776:20756] [Mapbox] [Warning, config_service]: Update request 25 in progress
2022-10-07 18:08:50.835897+0000 xctest[4776:20754] [Mapbox] [Warning, config_service]: Unexpected HTTP response code is received: 401

Too long with no output (exceeded 10m0s): context deadline exceeded
```

/ref https://github.com/mapbox/mapbox-navigation-ios/pull/4183#discussion_r984919182

File: CHANGELOG.md (hunk 2 above)
Comment (1ec5) [reply CHANGELOG.md:13]: * Fixed an issue where the current road name label appeared even while the user is not traveling on a known road. ([#4191](https://github.com/mapbox/mapbox-navigation-ios/pull/4191))

File: CHANGELOG.md [hunk 3]
```diff
@@ -17,24 +22,30 @@
...
 * Fixed an issue where the shields in the instruction are using the style from last navigation session with the `NavigationMapView` injection used in the new session. ([#4197](https://github.com/mapbox/mapbox-navigation-ios/pull/4197))
 
-### Routing
-
-* Route refreshing now respects current user progress on a leg to reduce update size and improve updating longer routes. ([#4111](https://github.com/mapbox/mapbox-navigation-ios/pull/4111))
-* Added `RoutingProvider.refreshRoute(indexedRouteResponse:fromLegAtIndex:routeShapeIndex:legShapeIndex:completionHandler:)` to request a refresh starting from specified shape index. Updated `RouteLegProgress` and `RouteProgress` initializers to include shape indices, and added `RouteProgress.refreshRoute(with:at:legIndex:legShapeIndex:)` to apply partial route refresh. ([#4111](https://github.com/mapbox/mapbox-navigation-ios/pull/4111))
-
 ### Banners and guidance instructions
```
Comment (1ec5) [inline CHANGELOG.md:39]: Normally we put the UI-related sections before the Core Navigation–related sections, in a progression from most user-facing to least, but I guess nothing actually depends on this order.

**Comment by jill-cardamon** [comment]
> Tests still [deadlock](https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-ios/9729/workflows/33abd3f7-276c-4052-8177-92cb0ba7219c/jobs/89206) even with MapboxCommon v23.1.0-rc.2

Will include a message in the release notes.

**Review by 1ec5 — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (jill-cardamon) [reply CHANGELOG.md:39]: There were two `Routing` sections so I moved the changes to the first one. I change the order though.

**Pull request merged**